api/agentpb/agent.proto

```diff
@@ -60,6 +60,9 @@ message SetStateRequest {
     int64 max_query_log_size = 4;
   }
   map<string, BuiltinAgent> builtin_agents = 2;
+  // Scrape configuration for vmagent, that runs at pmm-agent,
+  // built and validated at a pmm-server.
+  bytes vmagent_scrape_config = 3;
```

Comment:
Why not string?

Reply:
to avoid unneeded  []byte-> string,string -> []byte conversions.

 Pmm-server renders config with yaml.marshal to []byte, pmm-agent writes this config to a file.

Reply:
Well. Many conversions are actually being optimised by the compiler. But is also not like we care about conversion performance there – that place is far from being a hot. What we care about is debug-ability: bytes will not be logged properly in debug/trace modes, for example.

But you know, I actually think we should use `AgentProcess vmagent = 3` there, and pass everything, including templated flags, environment, etc from the PMM Server (`type` should be invalid in this case, but everything else should be used). That will allow use to enable debug mode from the server, for example, or send different flags for different versions of pmm-agent/vmagent.

/cc @BupycHuk

Reply:
Ok, get it, its not hard to change for string. 

 Need to think how to implement it better, you rejected this suggestion, last time we've discussed it. If we can simply add new agentType for vmagent it solves a lot of problems (scrape config generation, configuration, error logging and etc). For instance, you would be able to manage it from pmm-admin for "debug mode".

Reply:
Let's talk about it on or after our call tomorrow

Reply:
added vmagent type.